compiler/src/dotty/tools/dotc/ast/Trees.scala

```diff
@@ -1578,6 +1580,8 @@ object Trees {
               this(this(this(x, block), handler), finalizer)
             case SeqLiteral(elems, elemtpt) =>
               this(this(x, elems), elemtpt)
+            case Inlined(call, bindings, expansion) if includeAllNodes =>
+              this(this(this(x, call), bindings), expansion)(using inlineContext(call))
```

Comment:
You can do this in your own, e.g. Metals owned, TreeAccumulator or TreeTraverser by having it as a case in the `apply` of your implementation.  Note, there's also TreeMap, which skips the call, so you might want that implemented.

`inlineContext` is just the identity in `untpd`, so you can skip it, and it's public in `tpd`, so you can access it.

Reply:
I was thinking about that, but we also need annotation trees and that means we have to add a bunch of cases and this because quite fragile. For all the member trees and inline.  I also was thinking that other tools might have issues with this, so it might be better to upstream this.

Reply:
I think it should be done in metals.  Changing the behavior of TreeAccumulator should not be done without a thorough analysis. I believe most phases would _not_ want to recurse into calls. So we should leave it as is and override in a subclass where we need something different.

Reply:
>  Changing the behavior of TreeAccumulator should not be done without a thorough analysis

It's not an issue in the current approach since I added a separate flag. The default behaviour is unchanged.

>  I believe most phases would not want to recurse into calls.

I checked on another PR and indeed that breaks compilation, which is expected. However, there is no sensible way to traverse the AST currently aside from implementing it for each tool that will need it, since you cannot be sure which nodes will be included and which will not. It is a bit arbitrary from the point of view of tooling and has already caused a bunch of bugs in metals.

This is why I though it might be useful to have it available in the compiler behind the flag, but we can close this PR if you feel it's not something worth including.